modules/appgw/.header.md

```diff
@@ -0,0 +1,560 @@
+# Palo Alto Networks Application Gateway Module for Azure
+
+A terraform module for deploying a Application Gateway and its components required for the VM-Series firewalls in Azure.
+
+## Usage
+
+In order to use module `appgw`, you need to deploy `azurerm_resource_group` and use module `vnet` as prerequisites.
+Then you can use below code as an example of calling module to create APP GW:
+
+```hcl
+module "appgw" {
+  source = "../../modules/appgw"
+
+  for_each = var.appgws
+
+  name                = each.value.name
+  public_ip_name      = each.value.public_ip_name
+  resource_group_name = local.resource_group.name
+  location            = var.location
+  subnet_id           = module.vnet[each.value.vnet_key].subnet_ids[each.value.subnet_key]
+
+  managed_identities = each.value.managed_identities
+  capacity           = each.value.capacity
+  waf                = each.value.waf
+  enable_http2       = each.value.enable_http2
+  zones              = each.value.zones
+
+  frontend_ip_configuration_name = each.value.frontend_ip_configuration_name
+  listeners                      = each.value.listeners
+  backend_pool                   = each.value.backend_pool
+  backends                       = each.value.backends
+  probes                         = each.value.probes
+  rewrites                       = each.value.rewrites
+  rules                          = each.value.rules
+  redirects                      = each.value.redirects
+  url_path_maps                  = each.value.url_path_maps
+
+  ssl_global   = each.value.ssl_global
+  ssl_profiles = each.value.ssl_profiles
+
+  tags       = var.tags
+  depends_on = [module.vnet]
+}
+```
+
+Below there are provided sample values for `appgws` map (in comments you can find also commands to create SSL/TLS certificats, if required):
+
+```hcl
```

Comment:
IMHO the previous README was better :D you not only had examples, but they had some introduction and comments. Now we only have a huuuuge block of code. For someone who's new to this module/terraform, it can be unreadable and discouraging :) 

I know the previous README was long, but this change is for worst IMO :D

Reply:
I rebuild the README, but I don't want to repeat all properties, which are in tables below examples. 
Please let me know, if that changed README is acceptable.